src/components/diagrams/diagram-common.js

```diff
@@ -0,0 +1,323 @@
+        case 'DANGLING_LINE':
+            return equipments.danglingLines;
+        case 'GENERATOR':
+            return equipments.generators;
+        case 'VSC_CONVERTER_STATION':
+            return equipments.vscConverterStations;
+        case 'LCC_CONVERTER_STATION':
+            return equipments.lccConverterStations;
+        case 'HVDC_LINE':
+            return equipments.hvdcLines;
+        case 'CAPACITOR':
+        case 'INDUCTOR':
+            return equipments.shuntCompensators;
+        case 'STATIC_VAR_COMPENSATOR':
+            return equipments.staticVarCompensators;
+        case 'TWO_WINDINGS_TRANSFORMER':
+        case 'TWO_WINDINGS_TRANSFORMER_LEG':
+        case 'PHASE_SHIFT_TRANSFORMER':
+            return equipments.twoWindingsTransformers;
+        case 'THREE_WINDINGS_TRANSFORMER':
+        case 'THREE_WINDINGS_TRANSFORMER_LEG':
+            return equipments.threeWindingsTransformers;
+        default: {
+            console.log('bad feeder type ', feederType);
+            return null;
+        }
+    }
+}
+
+export const useDiagram = () => {
+    const dispatch = useDispatch();
+    const diagramStates = useSelector((state) => state.diagramStates);
+    const studyUuid = useSelector((state) => state.studyUuid);
+
+    useEffect(() => {
+        syncDiagramStateWithSessionStorage(diagramStates, studyUuid);
+    }, [diagramStates, studyUuid]);
+
+    const openDiagramView = useCallback(
+        (id, type) => {
+            dispatch(openDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const togglePinDiagramView = useCallback(
+        (id, type) => {
+            dispatch(togglePinDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const minimizeDiagramView = useCallback(
+        (id, type) => {
+            dispatch(minimizeDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const closeDiagramView = useCallback(
+        (id, type) => {
+            dispatch(closeDiagram(id, type));
+        },
+        [dispatch]
+    );
+
+    const closeDiagramViews = useCallback(
+        (idsToRemove) => {
+            dispatch(closeDiagrams(idsToRemove));
+        },
+        [dispatch]
+    );
+
+    return [
+        openDiagramView,
+        minimizeDiagramView,
+        togglePinDiagramView,
+        closeDiagramView,
+        closeDiagramViews,
+    ];
+};
+
+export function getNameOrId(value) {
+    return value?.name ?? value?.id;
+}
+
+export function getSubstationNameOrId(value) {
+    return value?.substationName ?? value?.substationId;
+}
+
+export const NoSvg = { svg: null, metadata: null, error: null, svgUrl: null };
+
+// Compute the paper and svg sizes. Returns undefined if the preferred sizes are undefined.
+export const computePaperAndSvgSizesIfReady = (
+    isFullScreenActive,
+    svgType,
+    totalWidth,
+    totalHeight,
+    svgPreferredWidth,
+    svgPreferredHeight,
+    headerPreferredHeight
+) => {
+    if (
```

Comment:
you can replace it by `svgPreferredWidth != null &&
         headerPreferredHeight != null`
!= null check also undefined